diffdrr/renderers.py

```diff
@@ -90,21 +88,30 @@ def forward(
             )
 
         # Multiply by ray length such that the proportion of attenuated energy is unitless
-        raylength = (target - source + self.eps).norm(dim=-1)
+        raylength = (target - source + self.eps).norm(dim=-1).to(torch.float32)
         img *= raylength.unsqueeze(1)
         return img
 
 # %% ../notebooks/api/01_renderers.ipynb 8
 def _get_alphas(
-    source, target, origin, spacing, dims, eps, filter_intersections_outside_volume
+    source, target, inverse_affine, dims, eps, filter_intersections_outside_volume
 ):
     """Calculates the parametric intersections of each ray with the planes of the CT volume."""
+    inverse_affine = inverse_affine.to(source)
     # Parameterize the parallel XYZ planes that comprise the CT volumes
-    alphax = (torch.arange(dims[0] + 1).to(source) - 0.5) * spacing[0] + origin[0]
-    alphay = (torch.arange(dims[1] + 1).to(source) - 0.5) * spacing[1] + origin[1]
-    alphaz = (torch.arange(dims[2] + 1).to(source) - 0.5) * spacing[2] + origin[2]
+    alphax = torch.arange(dims[0] + 1).to(source) - 0.5
+    alphay = torch.arange(dims[1] + 1).to(source) - 0.5
+    alphaz = torch.arange(dims[2] + 1).to(source) - 0.5
 
     # Calculate the parametric intersection of each ray with every plane
+    source = (
+        torch.einsum("nab, n...b -> n...a", inverse_affine[:, :3, :3], source)
+        + inverse_affine[:, :3, 3][:, None]
+    )
+    target = (
+        torch.einsum("nab, n...b -> n...a", inverse_affine[:, :3, :3], target)
+        + inverse_affine[:, :3, 3][:, None]
+    )
```

Comment:
More clean to implement `affine` and `inverse_affine` as `diffdrr.pose.RigidTransform`s.